seastar-macros/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+#[proc_macro_attribute]
+#[cfg(not(test))] // Work around for rust-lang/rust#62127
+pub fn main(
+    args: proc_macro::TokenStream,
+    item: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let input = syn::parse_macro_input!(item as syn::ItemFn);
+    let args = syn::parse_macro_input!(args as syn::AttributeArgs);
+
+    let inputs = &input.sig.inputs;
+    let name = &input.sig.ident;
+    let body = &input.block;
+    let attrs = &input.attrs;
+
+    if input.sig.asyncness.is_none() {
+        let msg = "the async keyword is missing from the function declaration";
+        return syn::Error::new_spanned(input.sig.fn_token, msg)
```

Comment:
I've never written a proc macro so take this comment with a grain of salt.
What is `fn_token`, what does it span? Function name? signature? whole function?
`input.sig` should be possible to use it, and it sounds like it should span the signature (I may be mistaken of course), which is where the error should be shown.

Reply:
`fn_token` is just the `fn` token. Agreed, the error should span the entire signature.